pubnub-gson/src/main/java/com/pubnub/api/PubNub.java

```diff
@@ -0,0 +1,227 @@
+import com.pubnub.api.v2.entities.ChannelGroup;
+import com.pubnub.api.v2.entities.ChannelMetadata;
+import com.pubnub.api.v2.entities.UserMetadata;
+import com.pubnub.api.v2.subscriptions.Subscription;
+import com.pubnub.api.v2.subscriptions.SubscriptionSet;
+import com.pubnub.internal.PubNubImpl;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.Set;
+
+public interface PubNub extends BasePubNub<
+        EventListener,
+        Subscription,
+        Channel,
+        ChannelGroup,
+        ChannelMetadata,
+        UserMetadata,
+        SubscriptionSet,
+        StatusListener> {
+
+    /**
+     * Initialize and return an instance of the PubNub client.
+     * @param configuration the configuration to use
+     * @return the PubNub client
+     */
+    static PubNub create(PNConfiguration configuration) {
+        return new PubNubImpl(configuration);
+    }
+
+    @NotNull PNConfiguration getConfiguration();
+
+    @NotNull SubscribeBuilder subscribe();
+
+    @NotNull UnsubscribeBuilder unsubscribe();
+
+    @NotNull PresenceBuilder presence();
+
+    @NotNull AddChannelsToPush addPushNotificationsOnChannels();
+
+    @NotNull RemoveChannelsFromPush removePushNotificationsFromChannels();
+
+    @NotNull RemoveAllPushChannelsForDevice removeAllPushNotificationsFromDeviceWithPushToken();
+
+    @NotNull ListPushProvisions auditPushChannelProvisions();
+
+    @NotNull WhereNow whereNow();
+
+    @NotNull HereNow hereNow();
+
+    @NotNull Time time();
+
+    @NotNull History history();
+
+    @NotNull FetchMessages fetchMessages();
+
+    @NotNull DeleteMessages deleteMessages();
+
+    @NotNull MessageCounts messageCounts();
+
+    @NotNull Grant grant();
+
+    @NotNull GrantTokenBuilder grantToken();
+
+    @NotNull
+    GrantTokenBuilder grantToken(Integer ttl);
+
+    @NotNull RevokeToken revokeToken();
+
+    @NotNull GetState getPresenceState();
+
+    @NotNull SetState setPresenceState();
+
+    @NotNull Publish publish();
+
+    @NotNull Signal signal();
+
+    @NotNull ListAllChannelGroup listAllChannelGroups();
+
+    @NotNull AllChannelsChannelGroup listChannelsForChannelGroup();
+
+    @NotNull AddChannelChannelGroup addChannelsToChannelGroup();
+
+    @NotNull RemoveChannelChannelGroup removeChannelsFromChannelGroup();
+
+    @NotNull DeleteChannelGroup deleteChannelGroup();
+
+    SetUUIDMetadata setUUIDMetadata();
+
+    @NotNull GetAllUUIDMetadata getAllUUIDMetadata();
+
+    @NotNull GetUUIDMetadata getUUIDMetadata();
+
+    @NotNull RemoveUUIDMetadata removeUUIDMetadata();
+
+    SetChannelMetadata.Builder setChannelMetadata();
+
+    @NotNull GetAllChannelsMetadata getAllChannelsMetadata();
+
+    @NotNull GetChannelMetadata.Builder getChannelMetadata();
+
+    RemoveChannelMetadata.Builder removeChannelMetadata();
+
+    @NotNull GetMemberships getMemberships();
+
+    @NotNull SetMemberships.Builder setMemberships();
```

Comment:
When in java code (e.g. integTest) I have
`pubNub.setMemberships()`
then I click on setMemberships() method to get more info about this method I don't see java doc being displayed.
I know that we don't have such description on master but maybe that is something to think of.
What do you think?
Maybe we could add it to modernisation todo list.

Reply:
added all Java PubNub client docs